pkg/controller/controller_utils.go

```diff
@@ -335,3 +338,17 @@ func (rt *RequestTracker) SetRequests(requests int) *RequestTracker {
 func (rt *RequestTracker) GetError() error {
 	return rt.err
 }
+
+// MergeConfigMap is a MergeFn for generic configmap.
+// parameter types are set to runtime.Object to match the signature of MergeFn
+func MergeConfigMap(existing, desired runtime.Object) error {
+	existingCm := existing.(*corev1.ConfigMap)
```

Comment:
Should we check this type assertion?

Reply:
I will inline this function to the `CreateOrUpdateConfigMap` interface for simplicity, a public method without type assertion checking is dangerous indeed.